mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/MlflowAutologEventPublisher.scala

```diff
@@ -43,14 +44,39 @@ private[autologging] trait MlflowAutologEventPublisherImpl {
   }
 
   // Exposed for testing
-  private[autologging] def getSparkDataSourceListener: SparkListener = {
-    // Get SparkContext & determine if REPL id is set - if not, then we log irrespective of repl
-    // ID, but if so, we log conditionally on repl ID
+  private[autologging] val databricksConfigProviderClassName =
+    "com.databricks.config.DatabricksClientSettingsProvider"
+
+  /**
+   * @returns True if Spark is running in a REPL-aware context. False otherwise.
+   */
+  private def isInReplAwareContext: Boolean = {
+    // Attempt to fetch the `spark.databricks.replId` property from the Spark Context.
+    // The presence of this ID is a clear indication that we are in a REPL-aware environment
     val sc = spark.sparkContext
     val replId = Option(sc.getLocalProperty("spark.databricks.replId"))
-    replId match {
-      case None => new SparkDataSourceListener(this)
-      case Some(_) => new ReplAwareSparkDataSourceListener(this)
+    if (replId.isDefined) {
+      return true
+    }
+
+    // If the `spark.databricks.replId` is absent, we may still be in a Databricks environment,
+    // which is REPL-aware. To check, we look for the presence of a Databricks-specific
+    // configuration class on the classpath
+    val configProviderFetchResult = Try {
+      Class.forName(databricksConfigProviderClassName);
+    }
```

Comment:
Ideally, we'd check for the presence of `getReplId()` in the Databricks entrypoint Java object. However, it was not immediately clear that this is resolvable from a Databricks Scala REPL. Input would be great here!

Reply:
cc @WeichenXu123 @smurching

Reply:
@WeichenXu123 We don't need / should not pass the repl ID to `ReplAwareSparkDataSourceListener`, since the listener needs to pull REPL ID information from Spark event properties. The listener fetches events from multiple REPLs.

Reply:
I'll wait for Bago's confirmation but I think it makes sense that the entrypoint is difficult to access from Scala code here - the entrypoint is a Python-REPL-specific interface. Another way to check that we're in Databricks could be to check for the `spark.databricks.clusterUsageTags.clusterId` property (`spark.conf.get("spark.databricks.clusterUsageTags.clusterId")`), which I believe is set prior to cluster startup (it's how we fetch cluster ID to set as a tag for the "Reproduce Run" feature), but we should check/confirm. It may be more stable than relying on a particularly-named class - either way I guess we can add a test for the API we end up relying on to detect that we're in Databricks from this Scala code

Reply:
Thanks for the suggestion @smurching ! I've gone ahead and replaced the logic that searches for `com.databricks.config.DatabricksClientSettingsProvider` with the proposed logic for fetching the `spark.databricks.clusterUsageTags.clusterId` conf. I've also updated tests accordingly.